code/datums/mind.dm

```diff
@@ -251,18 +252,26 @@ datum/mind
 		return FALSE
 
 	/// Attempts to remove existing antagonist datums of ID role_id from this mind.
-	proc/remove_antagonist(role_id)
-		for (var/datum/antagonist/A as anything in src.antagonists)
-			if (A.id == role_id)
-				A.remove_self(TRUE)
-				src.antagonists.Remove(A)
-				if (!length(src.antagonists) && src.special_role == A.id)
-					src.special_role = null
-					ticker.mode.traitors.Remove(src)
-					ticker.mode.Agimmicks.Remove(src)
-				qdel(A)
-				src.current.antagonist_overlay_refresh(TRUE, FALSE)
-				return TRUE
+	proc/remove_antagonist(role)
+		var/list/datum/antagonist/antagonist_roles = list()
```

Comment:
Seems you were preventing the same kind being added more than once in add_antagonist(), if you're only expecting one antagonist role in the list later on, maybe no need for a list?

Reply:
Preventing more than one being added, provided the new role's and existing role's pseudo status is the same. A mind may be in possession of two of the same antagonist types; one real, one pseudo.
I am still debating whether this is the best way to handle this, or whether it would be best for a new antagonist status to override the previous one, as it does with subordinate antagonists.